src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -52,5 +88,37 @@ public class ConcurrencyManager : IConcurrencyManager
          var tt = results.Values;
          return results;
       }
+
+      private async Task<ConcurrencyManagerResult<TResult>> returnWithExceptionHandling<TData, TResult>
+      (
+         int coreId, 
+         CancellationToken cancellationToken, 
+         Func<int, CancellationToken, TData, Task<TResult>> task, 
+         TData args, Func<TData, string> id
+      ) where TResult : class
+      {
+         TResult simulationResult;
+         try
+         {
+            simulationResult = await task.Invoke(coreId, cancellationToken, args);
+         }
+         catch (Exception e)
+         {
+            return new ConcurrencyManagerResult<TResult>
+            (
+               id(args),
+               false,
+               null,
+               e.Message
+            );
+         }
+         return new ConcurrencyManagerResult<TResult>
+         (
+            id(args),
+            true,
+            simulationResult,
```

Comment:
It feels like two constructor. One with Id(artgs) results and there other one just with the id(args)and error instead of asking the caller to pass the right parameters. (e.g. "", false or null true etc..)